src/components/button/__tests__/CdrButton.spec.js

```diff
@@ -3,113 +3,135 @@ import CdrButton from '../CdrButton.vue';
+    })
+
+    describe('with prop "type" set to "reset"', () => {
+      beforeEach(() => {
+        wrapper.setProps({ type: 'reset' });
+      });
+      it('sets type attr correctly', () => {
+        expect(wrapper.attributes().type).toBe('reset');
+      });
+    })
+
+    describe('when "el" is set to "a"', () => {
+      beforeEach(() => {
+        wrapper.setProps({ el: 'a' });
+      });
+
+      it('computes base class correctly', () => {
+        expect(wrapper.classes()).toContain('cdr-button');
+      });
+    })
+
+    describe('when "iconOnly" is set to true and size is "large"', () => {
+      beforeEach(() => {
+        wrapper.setProps({
+          size: 'large',
+          iconOnly: true,
+        });
+      });
+
+      it('adds custom size class', () => {
+        expect(wrapper.vm.sizeClass).toBe('cdr-button--icon-only-large');
+      });
+    })
+
+    describe('when "iconOnly" and "withBackground" is set to true', () => {
+      beforeEach(() => {
+        wrapper.setProps({
+          iconOnly: true,
+          withBackground: true,
+        });
+      });
+
+      it('adds class for icon only', () => {
+        expect(wrapper.classes()).toContain('cdr-button--icon-only');
+      });
+
+      it('adds class for icon background', () => {
+        expect(wrapper.classes()).toContain('cdr-button--with-background');
+      });
+    })
+
+    describe('with tag prop set to "a"', () => {
+      beforeEach(() => {
+        wrapper.setProps({ tag: 'a' });
+      });
+
+      it('renders a link', () => {
+        expect(wrapper.element.tagName).toBe('A');
+      });
+    })
+
+    describe('with a function passed to the onClick attribute', () => {
+      let spy;
+      beforeEach(() => {
+        spy = sinon.spy();
+        wrapper = shallowMount(CdrButton, {
+          attrs: {
+            onClick: spy
+          },
+        });
+        wrapper.trigger('click');
+      });
+      it('the function is called', () => {
+        expect(spy.calledOnce).toBeTruthy();
+      });
+    })
+
+
+
+    describe('with an icon-left slot', () => {
+      beforeEach(() => {
+        wrapper = shallowMount(CdrButton, {
+          slots: {
+            default: ['default slot for text'],
+            'icon-left': '<div />',
+          },
+        });
+      });
+      it('adds the "cdr-button--has-icon-left" class', ()=>{
+        expect(wrapper.classes()).toContain('cdr-button--has-icon-left');
+      })
     });
 
-    expect(wrapper.classes()).toContain('cdr-button--has-icon-right');
-  });
-
-  it('renders a link', () => {
-    const wrapper = shallowMount(CdrButton, {
-      propsData: {
-        tag: 'a',
-      },
+    describe('with an icon-left slot', () => {
+      beforeEach(() => {
+        wrapper = shallowMount(CdrButton, {
+          slots: {
+            default: ['default slot for text'],
+            'icon-right': '<div />',
+          },
+        });
+      });
+      it('adds the "cdr-button--has-icon-right" class', ()=>{
```

Comment:
Should we do a snapshot test here, instead of testing for just one class? That way we can account for any other outliers.

There are a lot of tests that are testing for one specific attribute or class. Maybe a snapshot test should be added in addition to these. While this is redundant, coverage-wise, sometimes specific expect statements can be useful in calling out what's wrong so that we don't have to dig into the snapshot comparison every time when there's a failing unit test